test/net_http_test.rb

```diff
@@ -312,6 +313,44 @@ def test_adding_custom_errors_do_trip_circuit
     end
   end
 
+  def test_5xxs_trip_circuit_when_fatal_server_flag_enabled
+    semian_config = {
+      tickets: 3,
+      success_threshold: 1,
+      error_threshold: 3,
+      error_timeout: 10,
+      open_circuit_server_errors: true,
+    }
+    modified_semian_config = proc do |host, port|
+      next nil if host == "127.0.0.1" && port == 8474 # disable if toxiproxy
+      semian_config.merge(name: "#{host}_#{port}")
+    end
```

Comment:
Could you make this part of the `with_semian_configuration` helper as a default, so we don't duplicate the Toxiproxy stuff? Feels really dirty 😬